### PR TITLE
[DOCS] Adds description to the preview data frame transform API

### PR DESCRIPTION
**Pull request by szabosteve**
This PR adds a description section to the preview data frame transform API.

### CODE REVIEW

File: docs/reference/data-frames/apis/preview-transform.asciidoc [hunk 1]
```diff
@@ -40,6 +49,13 @@ Previews a {dataframe-transform}.
 `pivot` (Required)::
   (object) Defines the pivot function `group by` fields and the aggregation to
   reduce the data. See <<data-frame-transform-pivot>>.
+  
+[discrete]
+[[preview-data-frame-transform-response]]
+==== {api-response-body-title}
+
+`preview`::
+  (array) An array of {dataframe} objects.
```
Comment (lcawl) [inline docs/reference/data-frames/apis/preview-transform.asciidoc]: @hendrikmuhs Is this an appropriate description? If not, what would you recommend calling the contents of the array?

Comment (benwtrent) [reply docs/reference/data-frames/apis/preview-transform.asciidoc]: `An array of documents` is more apt. 

They are the JSON representation of the documents that will be indexed into the destination of the transform. They are not {dataframe} objects.

Comment (lcawl) [reply docs/reference/data-frames/apis/preview-transform.asciidoc]: Thanks @benwtrent I've revamped the description

**Review by lcawl — APPROVED** [review]
LGTM

**Comment by szabosteve** [comment]
Merged and backported to 7.2, 7.3, and 7.x.

**Pull request merged**